include/swift/AST/DiagnosticsSema.def

```diff
@@ -8592,8 +8592,8 @@ ERROR(value_type_used_in_type_parameter,none,
       "cannot use value type %0 for generic argument %1", (Type, Type))
 ERROR(invalid_value_type_value_generic,none,
       "%0 is not a supported value type for %1", (Type, Type))
-ERROR(invalid_value_generic_conformance,none,
-      "value generic type %0 cannot conform to protocol %1", (Type, Type))
+ERROR(invalid_value_generic_constraint,none,
+      "inheritance constraint for value generic type %0 has no meaning", (Type))
```

Comment:
I would prefer the collective term to be "subtype constraint". I chose "inheritance constraint" because we already sort of exercise this wording in `candidate_types_inheritance_requirement`.

Reply:
Also not a big fan of "value generic type". I think it should be "value generic parameter".

Reply:
Alternatively, maybe we go more broad and just say `cannot constrain value generic parameter`? Or maybe `cannot use type constraint with value generic parameter`?

Reply:
I mildly prefer the emphasis on "this has no meaning" as a formal way to say "this does not make sense fundamentally" vs. "cannot do this". But happy to switch to "type constraint".

Reply:
I guess I have a slight aversion to saying this makes no sense, because we could always _choose_ to interpret it in a way that does make sense (the constraint would apply to the value type), albeit it would be redundant. I think I have a mild preference for `cannot use type constraint with value generic parameter`, I quite like the fact that it's explicitly contrasting "type constraint" with "value".